iso15118/secc/states/iso15118_20_states.py

```diff
@@ -1450,7 +1476,66 @@ async def process_message(
         ],
         message_exi: bytes = None,
     ):
-        raise NotImplementedError("DCCableCheck not yet implemented")
+        msg = self.check_msg_v20(message, [DCCableCheckReq, SessionStopReq], False)
+        if not msg:
+            return
+
+        if isinstance(msg, SessionStopReq):
+            await SessionStop(self.comm_session).process_message(message, message_exi)
+            return
+
+        dc_cable_check_req: DCCableCheckReq = msg  # noqa
+
+        if not self.cable_check_req_was_received:
+            # Requirement in 6.4.3.106 of the IEC 61851-23
+            # Any relays in the DC output circuit of the DC station shall
+            # be closed during the insulation test
+            if not await self.comm_session.evse_controller.is_contactor_closed():
+                self.stop_state_machine(
+                    "Contactor didnt close for Cable Check",
+                    message,
+                    ResponseCode.FAILED,
+                )
+                return
+            await self.comm_session.evse_controller.start_cable_check()
```

Comment:
as I mentioned to you before, we need to invert this. First we start cable check, then we check if the contactors are closed

Reply:
I will approve the PR, so I dont block your work, but change that if you can ;)

Reply:
Yes have created a ticket [here](https://dev.azure.com/switch-ev/Josev/_backlogs/backlog/Josev%20Team/Epics/?workitem=3669).